fix(property-select): trim whitespace from select options in PropertyFilters

lawalbolaji opened the pull request:
Fixes #17254 - Cohorts property ID filter copy and pasting values bug

## Problem
If you try to create a cohort based on multiple values for filtering criteria e.g. organization_id, and then paste the list of values into the property filter, the values appear to paste properly in the field, but the cohort persons return based on the first ID only.

This is because of how the multi-select component handled the tokenization of the values. It expected a "comma" separated list in the strict sense so token1, token2, token3... was converted to "token1", " token2", "token3" etc.

## Changes
Trimmed the tokenized property filters after pasting.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?
Manually:
- Paste a comma-separated list of values with spaces into the cohort filter
- Confirm that the cohort matches all of the specified values

lawalbolaji committed:
fix copy and paste bug for comma separated filter ids

neilkakkar commented:
This is great, thanks for the contribution @lawalbolaji :)

neilkakkar commented:
Hey @lawalbolaji I just tested this out, it works great, thanks!

One thing worth fixing here as well: This holds for the multiselect, but there's a similar problem in the single select (specifically here: https://github.com/PostHog/posthog/blob/master/frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx#L209 ) where if I choose the operator as `contains` , and then have a space at the end, it tries to match with the space, which breaks.

Do you want to fix this too in your PR?

liyiy reviewed (approved):
This PR LGTM, as I cannot think of a reason why people would want trailing whitespace in their values 😃 

+1 to @neilkakkar's comment above though while we're on the same PR

lawalbolaji committed:
fix: strip whitespace from cohort property field;

- also allow user to navigate through autocomplete suggestions (up+down arrow keys) and choose with the enter key.

lawalbolaji committed:
Merge branch 'master' into fix/17254-cohorts-property-filter-bug

lawalbolaji commented:
> Hey @lawalbolaji I just tested this out, it works great, thanks!
> 
> One thing worth fixing here as well: This holds for the multiselect, but there's a similar problem in the single select (specifically here: https://github.com/PostHog/posthog/blob/master/frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx#L209 ) where if I choose the operator as `contains` , and then have a space at the end, it tries to match with the space, which breaks.
> 
> Do you want to fix this too in your PR?

Sure @neilkakkar I've updated the pr to fix that. One more thing worth noting here though, I noticed that when you type a value in the single select field and then hit enter (key) on one of the autocomplete suggestions, it doesn't work - I fixed that as well.

lawalbolaji commented on frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx:
```diff
@@ -205,19 +205,10 @@ export function PropertyValue({
                 setInput('')
                 setValue('')
             }}
-            onChange={(val) => {
```
There are 3 primary change events that happen with this autoComplete component - onSearch (user types into the box), onChange (user types into the box or selects an autocomplete suggestion) and onSelect (user selects autocomplete suggestion).

In all cases, onSearch and onSelect fire after onChange, and because react batches state updates, the state update in the onChange handler is useless, since the same state is updated in the onSearch and onSelect handlers.

lawalbolaji committed:
fix: revert onChange handler

lawalbolaji commented on frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx:
```diff
@@ -109,9 +109,9 @@ export function PropertyValue({
 
     const commonInputProps = {
         onSearch: (newInput: string) => {
-            setInput(newInput)
```
The change needs to happen here instead of [line209](https://github.com/lawalbolaji/posthog/blob/9f839d3ae30363c4faa73e16f5cc8763e65cf50b/frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx#L209) for 2 reasons:

- The search event fires **after** the change event, and because react will batch the state updates, any state modification in the onChange handler is basically useless.
- I couldn't find a scenario where user input did not fire the search event, and even when users select an autocomplete suggestion, the select event is handled by the onSelect handler.

I would go as far as ditching the onChange handler but the docs insist on using it for controlled inputs... even though onSelect and onSearch are called right after, depending on context.

neilkakkar replied:
A somewhat annoying side effect of this solution is that if I have a space in the filter (in between letters) I can't type it anymore, so for example if I wanted to type a[space]c, I'll need to type `ac` and then add a space in between, as the `trim()` doesn't let me input `a[space]`.

But, this problem is better than the current one, so I'll merge this as is, but feel free to fix this in a follow up :)

neilkakkar commented:
Thanks @lawalbolaji for the contribution :) 

Send me an email at neil [at] posthog.com and I'd love to send you a posthog merch code for your contribution :)
